gui/system/forms.py

```diff
@@ -2705,6 +2717,19 @@ class Meta:
         model = models.Certificate
 
 
+def is_valid_privatekey(certificate, privatekey):
+    pkey_obj = crypto.load_privatekey(crypto.FILETYPE_PEM, privatekey)
+    cert_obj = crypto.load_certificate(crypto.FILETYPE_PEM, certificate)
+    ctx = SSL.Context(SSL.TLSv1_METHOD)
```

Comment:
I tested this against the kind of certificates which get generated from freenas gui. It works for both CA and certificates. No particular reason as such.
Would it get restricted for other digital certificates?